src/wmtk/io/MeshReader.cpp

```diff
@@ -71,7 +71,9 @@ void MeshReader::set_attribute(
 
     accessor.set_attribute(v);
 
-    mesh.set_capacities_from_flags();
+    if (name == "flags") {
```

Comment:
perhaps we need a `set_capacitity_from_flag(PrimitiveType)` here. This current implementation will reload ALL capacities every time

Reply:
Yes but the overhead is really minor. Eventually, we won't have more than four capacities.